nisystemlink/clients/spec/utilities/_dataframe_utilities.py

```diff
@@ -100,6 +95,53 @@ def convert_specs_to_dataframe(
     return specs_dataframe
 
 
+def __convert_spec_to_dict(
+    spec: Specification, condition: Dict[str, Any]
+) -> Dict[str, Any]:
+    """Converts a spec into dictionary.
+
+    Args:
+        spec: Spec object.
+        condition: Condition as a dictionary which is added to the output spec dictionary.
+
+    Returns:
+        Spec as a dictionary with the provided condition dictionary included.
+    """
+    return {
+        **{
+            key: value
+            for key, value in vars(spec).items()
+            if key not in ["type", "limit", "conditions"]
+        },
+        **(__serialize_type(spec.type) if spec.type else {}),
+        **(__serialize_limits(spec.limit) if spec.limit else {}),
+        **{key: value for key, value in condition.items()},
+    }
+
+
+def __default_condition_formatting(
```

Comment:
Similar to the summarize case, our out of the box formatting should be public functions. This allows the caller to explicitly and declaratively opt in to the default or take the default function as a baseline they work on top of. That should also mean we give this a more descriptive name, like `condition_per_row`

Reply:
Or in light of the other comments, `condition_per_column`